===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ p11_kit_pin_register_callback:n$1 == { 0 } }) free(*(p11_kit_pin_register_callback:cb)) at 3 (line 252, column 9) }
===== Actual patch in diff format =====
@@ -248,7 +249,8 @@ p11_kit_pin_register_callback (const char *pin_source,
 	ret = register_callback_unlocked (pin_source, cb);
 
 	p11_unlock ();
-
+	if (ret == 0)
+		free(cb);
 	return ret ? 0 : -1;
 }
 
